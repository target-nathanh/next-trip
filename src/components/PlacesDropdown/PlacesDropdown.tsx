import React from 'react';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import find from 'lodash/find';
import { Place } from '../../types';

interface PlacesDropdownProps {
  places: Place[];
  selectedPlace?: Place;
  onSelectPlace: (place: Place) => void;
}

const PlacesDropdown: React.FC<PlacesDropdownProps> = (props: PlacesDropdownProps) => {
  const onChange = (event: React.ChangeEvent<{ value: unknown }>) => {
    const selectedPlace = find(props.places, { place_code: event.target.value }) as Place;
    if (selectedPlace) {
      props.onSelectPlace(selectedPlace);
    }
  };
  return (
    <Select value={props.selectedPlace?.place_code} onChange={onChange}>
      {props.places?.map((place: Place) => (
        <MenuItem key={place.place_code} value={place.place_code}>
          {place.description}
        </MenuItem>
      ))}
    </Select>
  );
};

export default PlacesDropdown;
