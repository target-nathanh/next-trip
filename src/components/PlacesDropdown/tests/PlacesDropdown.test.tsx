import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import places from '../../../mockData/placesMock.json';
import PlacesDropdown from '../PlacesDropdown';
import { MemoryRouter } from 'react-router-dom';

describe('PlacesDropdown', () => {
  test('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <PlacesDropdown
          places={places}
          selectedPlaceId={places[0].place_code}
          onSelectPlace={() => {}}
        />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('calls onSelectPlace when new place is selected', async () => {
    const onSelectPlaceMock = jest.fn();
    const { getByRole } = render(
      <MemoryRouter>
        <PlacesDropdown
          places={places}
          selectedPlaceId={places[0].place_code}
          onSelectPlace={onSelectPlaceMock}
        />
      </MemoryRouter>
    );
    fireEvent.mouseDown(getByRole('button'));

    const option = await screen.findByText(places[1].description);
    fireEvent.click(option);
    expect(onSelectPlaceMock).toBeCalledWith(places[1].place_code);
  });
});
