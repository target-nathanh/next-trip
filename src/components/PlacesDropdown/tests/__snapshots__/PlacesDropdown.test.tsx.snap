// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesDropdown matches snapshot 1`] = `
<div
  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-1"
>
  <div
    aria-haspopup="listbox"
    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    role="button"
    tabindex="0"
  >
    MSP Airport Terminal 1 - Lindbergh Station
  </div>
  <input
    aria-hidden="true"
    class="MuiSelect-nativeInput"
    placeholder="Select stop"
    tabindex="-1"
    value="LIND"
  />
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
